ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala

```diff
@@ -169,16 +151,49 @@ object LedgerContext {
 
     require(esf != null, "ExecutionSequencerFactory must not be null.")
 
-    def ledgerId: String =
+    private val logger = LoggerFactory.getLogger(this.getClass)
+
+    val cachedLedgerId: domain.LedgerId =
```

Comment:
we should drop this I think and just write: `val ledgerId ..` instead. There is no caching anymore here, just initialisation

Reply:
Ah, good to know that overriding `def` with `val` is fine!